maths/series/p_series.py

```diff
@@ -1,48 +1,38 @@
 """
 This is a pure Python implementation of the P-Series algorithm
 https://en.wikipedia.org/wiki/Harmonic_series_(mathematics)#P-series
-
 For doctests run following command:
 python -m doctest -v p_series.py
 or
 python3 -m doctest -v p_series.py
-
 For manual testing run:
 python3 p_series.py
 """
 
 
-def p_series(nth_term: int, power: int) -> list:
-    """Pure Python implementation of P-Series algorithm
-
+def p_series(nth_term: int, power: int) -> list[float]:
+    """
+    Pure Python implementation of P-Series algorithm
     :return: The P-Series starting from 1 to last (nth) term
-
     Examples:
     >>> p_series(5, 2)
-    [1, '1/4', '1/9', '1/16', '1/25']
+    [1.0, 0.25, 0.1111111111111111, 0.0625, 0.04]
     >>> p_series(-5, 2)
     []
     >>> p_series(5, -2)
-    [1, '1/0.25', '1/0.1111111111111111', '1/0.0625', '1/0.04']
```

Comment:
This looks like a cooler test.

Reply:
Make all strings here instead of floats, then?

Reply:
The current implementation is tolerant of both.  We should not lose that flexibility.  Our type hint says that we _expect_ float but our implementation is flexible to do the right thing if someone hands us something that is not a float but is still reasonable.

Reply:
https://pydantic-docs.helpmanual.io does a nice job of this.